caravel/models.py

```diff
@@ -673,18 +674,25 @@ def query(  # sqla
                 select_exprs += [timestamp_grain]
                 groupby_exprs += [timestamp_grain]
 
-            tf = '%Y-%m-%d %H:%M:%S.%f'
+            column_info = db.session().query(TableColumn).filter_by(
```

Comment:
no need to re-query the db here, there's a dict named `cols` in scope here, so `dttm_col = cols.get(granularity)`. You can do that earlier in this method and use `dttm_col` at line 661 as well.

Assuming `col.dttm_sql_literal` will simplify the rest of the changes here.